modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java

```diff
@@ -286,6 +290,120 @@ public CalciteContextException newValidationError(
         return newEx;
     }
 
+    @Override
+    protected void checkTypeAssignment(
```

Comment:
can you add comment why this method was introduced and/or what is the difference with the original `SqlValidator.checkTypeAssignment` ?

Reply:
commend has been added